src/deep-security/lib/CredentialsManager.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Created by mgoria on 11/6/15.
+ */
+
+'use strict';
+
+import AWS from 'aws-sdk';
+import CognitoSyncManager from 'amazon-cognito-js';
+import {CreateCognitoDatasetException} from './Exception/CreateCognitoDatasetException';
+import {PutCognitoRecordException} from './Exception/PutCognitoRecordException';
+import {SynchronizeCognitoDatasetException} from './Exception/SynchronizeCognitoDatasetException';
+
+export class CredentialsManager {
+  /**
+   * @returns {String}
+   */
+  static get DATASET_NAME() {
+    return 'deep_session';
+  }
+
+  /**
+   * @returns {String}
+   */
+  static get RECORD_NAME() {
+    return 'session_creds';
+  }
+
+  /**
+   * @param {String} identityPoolId
+   * @param {Object|null} cognitoSyncClient
+   */
+  constructor(identityPoolId, cognitoSyncClient = null) {
+    this._identityPoolId = identityPoolId;
+    this._cognitoSyncClient = cognitoSyncClient;
+  }
+
+  /**
+   * @returns {Object}
+   */
+  get cognitoSyncClient() {
+    if (!this._cognitoSyncClient) {
+      this._cognitoSyncClient = new AWS.CognitoSyncManager();
+    }
+
+    return this._cognitoSyncClient;
+  }
+
+  /**
+   * @param {Object} credentials
+   * @param {Function} callback
+   */
+  saveCredentials(credentials, callback) {
+    this._createOrGetDataset((error, dataset) => {
+      if (error) {
+        return callback(error);
+      }
+
+      dataset.put(CredentialsManager.RECORD_NAME, this._encodeCredentials(credentials), (error, record) => {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['record' is defined but never used.](https://www.codacy.com/app/MitocGroup/deep-framework/file/1363148751/issues/source?bid=2895475&fileBranchId=2973909#l58)